PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessClusterUpdateDialog.java

```diff
@@ -0,0 +1,61 @@
+package com.microsoft.azure.sparkserverless.serverexplore.ui;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessCluster;
+import com.microsoft.azure.sparkserverless.serverexplore.SparkServerlessClusterUpdateCtrlProvider;
+import com.microsoft.azure.sparkserverless.serverexplore.sparkserverlessnode.SparkServerlessADLAccountNode;
+import com.microsoft.azure.sparkserverless.serverexplore.sparkserverlessnode.SparkServerlessClusterNode;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.intellij.rxjava.IdeaSchedulers;
+
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.Arrays;
+
+public class SparkServerlessClusterUpdateDialog extends SparkServerlessProvisionDialog {
+
+    private void setFieldsUneditable() {
+        Arrays.asList(
+                clusterNameField, sparkEventsField, masterCoresField, masterMemoryField, workerCoresField, workerMemoryField)
+                .forEach(field -> field.setEditable(false));
+        sparkVersionComboBox.setEditable(false);
```

Comment:
A special one?

Reply:
Yes! We can't put sparkVersionComboBox into the array since the they use different implementations. You can check `JTextComponent.setEditable` and `JComboBox.setEditable`

---
In reply to: [197366117](https://github.com/Microsoft/azure-tools-for-java/pull/1734#discussion_r197366117) [](ancestors = 197366117)